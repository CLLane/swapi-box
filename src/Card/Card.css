.card {
  border-radius: 8px;
  width: 300px;
  height: 250px;
  text-align: center;
  color: white;
  margin-left: 20px;
  margin-top: 30px;
  text-decoration: none;
  background-color: rgba(0, 0, 0, .5)
}

.card:hover {
     box-shadow:
    0 0 0.1vw  0.3vw #fff7f7,   
    0 0 0.4vw  0.4vw #FFE81F, 
    0 0   1.5vw  0.3vw #faee7c,
    inset 0 0 1.5vw  0.3vw rgb(250, 238, 124),
    inset 0 0 0.4vw  0.1vw #FFE81F,
    inset 0 0 0.5vw  0.1vw #fff7f7;
    background-color: rgba(0, 0, 0, .9)
}

h1 {
  padding-top: 50px;
}

.favorite-img {
  height: 40px;
  width: 40px;
}