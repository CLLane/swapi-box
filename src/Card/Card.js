import React from 'react';
import './Card.css';

const Card = ({ name, homeworld, species, population }) => {

  return (
    <div>
      <h1>{name}</h1>
      <p>{homeworld}</p>
    </div>
  )
};

export default Card;