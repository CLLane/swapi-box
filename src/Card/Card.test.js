import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {

  it('should match the snapshot with all data passed in correctly', () => {
    const residents = []
    const wrapper = shallow(<Card
      id={1}
      name='Luke Skywalker'
      homeworld= 'Tatooine'
      species= 'Human'
      population= {200000}
      terrain= 'grasslands, mountains'
      climate= 'temperate'
      residents={residents}
      model= 'Digger Crawler'
      vehicleClass= 'wheeled'
      numberOfPassengers= '30'
      toggleFav={jest.fn()}
      type = '/people'
      />);

      expect(wrapper).toMatchSnapshot(); 
  })

  it('should match the snapshot if the card is favorited', () => {
    const wrapper = shallow(<Card
      id={1}
      name='Luke Skywalker'
      homeworld= 'Tatooine'
      species= 'Human'
      population= {200000}
      terrain= 'grasslands, mountains'
      climate= 'temperate'
      model= 'Digger Crawler'
      vehicleClass= 'wheeled'
      numberOfPassengers= '30'
      isFavorite={true}
      toggleFav={jest.fn()}
      type = '/people'
      />);

      expect(wrapper).toMatchSnapshot(); 
  })

  it('should call the toggleFav prop with the card\'s id when clicked', () => {
    const toggleFavMock = jest.fn();
    const wrapper = shallow(<Card
      id={1}
      name='Luke Skywalker'
      homeworld= 'Tatooine'
      species= 'Human'
      population= {200000}
      terrain= 'grasslands, mountains'
      climate= 'temperate'
      model= 'Digger Crawler'
      vehicleClass= 'wheeled'
      numberOfPassengers= '30'
      isFavorite={true}
      toggleFav={toggleFavMock}
      type = '/people'
      />); 
      wrapper.find('.favorite-img').simulate('click');

      expect(toggleFavMock).toHaveBeenCalled();
  })
});