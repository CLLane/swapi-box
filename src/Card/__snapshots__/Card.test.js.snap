// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot if the card is favorited 1`] = `
<div>
  <section
    className="card"
  >
    <img
      alt=""
      className="favorite-img"
      onClick={[Function]}
      src="favorite.svg"
    />
    <h1
      className="small-cards-h1"
    >
      Luke Skywalker
    </h1>
    <h4
      className="small-card-labels"
    >
      Homeworld: 
      Tatooine
    </h4>
    <h4
      className="small-card-labels"
    >
      Species: 
      Human
    </h4>
    <h4
      className="small-card-labels"
    >
      Population: 
      200000
    </h4>
    <h4
      className="small-card-labels"
    >
      Terrain: 
      grasslands, mountains
    </h4>
    <h4
      className="small-card-labels"
    >
      Climate: 
      temperate
    </h4>
    <h4
      className="small-card-labels"
    >
      Model: 
      Digger Crawler
    </h4>
    <h4
      className="small-card-labels"
    >
      Vehicle Class: 
      wheeled
    </h4>
    <h4
      className="small-card-labels"
    >
      Number of Passengers: 
      30
    </h4>
  </section>
</div>
`;

exports[`Card should match the snapshot with all data passed in correctly 1`] = `
<div>
  <section
    className="card"
  >
    <img
      alt=""
      className="favorite-img"
      onClick={[Function]}
      src="isNotFavorite.svg"
    />
    <h1
      className="small-cards-h1"
    >
      Luke Skywalker
    </h1>
    <h4
      className="small-card-labels"
    >
      Homeworld: 
      Tatooine
    </h4>
    <h4
      className="small-card-labels"
    >
      Species: 
      Human
    </h4>
    <h4
      className="small-card-labels"
    >
      Population: 
      200000
    </h4>
    <h4
      className="small-card-labels"
    >
      Terrain: 
      grasslands, mountains
    </h4>
    <h4
      className="small-card-labels"
    >
      Climate: 
      temperate
    </h4>
    <h4
      className="small-card-labels"
    >
      Residents: 
    </h4>
    <h4
      className="small-card-labels"
    >
      Model: 
      Digger Crawler
    </h4>
    <h4
      className="small-card-labels"
    >
      Vehicle Class: 
      wheeled
    </h4>
    <h4
      className="small-card-labels"
    >
      Number of Passengers: 
      30
    </h4>
  </section>
</div>
`;
